tools/amalgamate.py

```diff
@@ -61,7 +62,7 @@
 #endif\n''' % (outname.upper() + u"_H_", outname.upper() + u"_H_") )
 
 for fname in args.inputs:
-    with io.open(fname, encoding="utf8") as infile:
+    with io.open(fname, encoding=None) as infile:
```

Comment:
Are you sure that the encoding hint in the first line also defines the encoding for the open command?

https://docs.python.org/2/library/io.html says that if encoding is not set, it is platform dependent.

I think on line 29 `file = io.open(args.outfile, 'w')` we should then also indicate the encoding?